nvtabular/ops/categorify.py

```diff
@@ -824,13 +824,22 @@ def _write_uniques(dfs, base_path, col_group, options):
                 if nlargest < len(df):
                     df = df.nlargest(n=nlargest, columns=name_count)
             if not _series_has_nulls(df[col]):
+                if name_count in df:
+                    df = df.sort_values(name_count, ascending=False, ignore_index=True)
```

Comment:
I think we might be sorting the values a couple times here unnecessarily: Is the sort_values on line 804 still necessary with this?  Also is this sort_values still necessary if we have a max_size parameter passed?

Reply:
Absolutely! The sort on 804 is done by the column itself, not the count. This sort brings the NA values first... it allows us to guarantee handling logic when a series does have nulls (we know its at the front and its only the first value). The second sort is by count. But we still need to maintain unknown as index 0.

Reply:
ok. For just ordering the NA value first we could probably do something a little cheaper than a full sort (like O(n) instead of O(n * log(N)) - but this works for now .